### ISSUE TITLE
[QUESTION] Series Scaling in historical_forecasts

### ISSUE THREAD

**Issue by Cajeux1999** [unit8co/darts#2738]
**Describe the issue linked to the documentation**

Hi,
I am working with a sliding-window cross-validation setup, with the method `historical_forecasts`.

In order to scale the series, i am doing as follows:
1. Separate series for y_train, y_val, past_cov_train, past_cov_val
2. Fit scaler for y_train and use the same scaler to transform y_val (same for past_cov).

```

scaler_y = Scaler()
scaler_air_temp = Scaler()

train_y = scaler_y.fit_transform(TimeSeries.from_series(train_data['y']))
past_cov_train = scaler_air_temp.fit_transform(TimeSeries.from_series(train_data['air_temp']))
val_y = scaler_y.transform(TimeSeries.from_series(val_data['y']))
past_cov_val = scaler_air_temp.transform(TimeSeries.from_series(val_data['air_temp']))
```

3. Concatenate train and val series in order to pass to historical_forecasts:
`merged_y = concatenate([train_y, val_y], axis=0)
`

4. Use the function, setting the validation start as the position in which y_val starts in the concatenated series:

`backtest = lgbm_model.historical_forecasts(merged_y, past_covariates=past_cov, forecast_horizon=h, start=val_start, train_length=train_length, last_points_only=False, stride=stride, retrain=True,)
`

So, my question is: Is there a way to include a Scaler() directly in the `historical_forecasts` method, so it handles the scaling as the window walks forward?

The pipeline i used tests the model and avoid leakage, but, the scaling part perhaps could be optimized directly in `historical_forecasts `.

**Comment by dennisbader** [comment]
Yes, you can use the `data_transformer` argument for that in methods like historical_forecasts(), backtest() and residuals(). See the docs for example [here](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.regression_model.html#darts.models.forecasting.regression_model.RegressionModel.historical_forecasts)

**Comment by Cajeux1999** [comment]
Hi Dennis,

Thanks for your answer.

I have a follow-up question:

If i have 3 `past_covariates `for the target series y, how can i scale them separately using `data_transformers` (if the covariates have different data ranges)?
```

scaler_y = Scaler()

covariate1_transformer = Scaler()
covariate2_transformer = Scaler()
covariate3_transformer = Scaler()

```
The documentation mentions only one key for past_covariates:

`Optionally, a dictionary of BaseDataTransformer or Pipeline to apply to the corresponding series (possibles keys; “series”, “past_covariates”, “future_covariates”)
`

How could i setup my dictionary of data_transformers?

```
data_transformers = {
    'series': series_transformer,
    'past_covariates': covariate1_transformer
...
}
```

**Comment by madtoinou** [comment]
Hi @Cajeux1999,

The `Scaler` class has a `global_fit` argument ([doc](https://unit8co.github.io/darts/generated_api/darts.dataprocessing.transformers.scaler.html?highlight=scaler#darts.dataprocessing.transformers.scaler.Scaler)), which indicate if it should apply one set of parameters per series or the same one on all of them.

Note that `historical_forecasts` expect one covariate for each series, it won't map 3 covariates series to one target series. You will need to stack the covariates series into a single series prior to passing them to the method (and then, the `Scaler` will be applied on each component independently).
